src/components/dev-dot-content/mdx-components/mdx-heading-permalink/index.tsx

```diff
@@ -3,20 +3,31 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
+import Link from 'next/link'
 import classNames from 'classnames'
 import { IconLink16 } from '@hashicorp/flight-icons/svg-react/link-16'
 import s from './mdx-heading-permalink.module.css'
 
-export default function MdxHeadingPermalink(props) {
-	const { className, level, href, 'aria-label': ariaLabel } = props
+interface MdxHeadingPermalinkProps {
+	className?: string
+	level: 1 | 2 | 3 | 4 | 5 | 6
+	href: string
+	ariaLabel: string
+}
 
+export default function MdxHeadingPermalink({
```

Comment:
I've updated `MdxHeadingPermalink` to use `Link`. This ensures more efficient client-side navigation, and ensures the interactions we expect with `next/router`. The `next/router` bit is important for monitoring the URL for `#hash` changes, which is a key part of our `#user-content-` hash workaround.

I also added types here for completeness.